crates/stages/src/stages/sender_recovery.rs

```diff
@@ -81,8 +79,8 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
         // Iterate over transactions in chunks
         info!(target: "sync::stages::sender_recovery", start_tx_index, end_tx_index, "Recovering senders");
 
-        // An _unordered_ channel to receive results from a rayon job
-        let (tx, rx) = mpsc::unbounded_channel();
+        // channels used to return result of sender recovery.
+        let mut channels = Vec::new();
```

Comment:
can alloc capacity with the `chunks` size

```
self.commit_threshold as usize / rayon::current_num_threads()
```

Reply:
Not that big of a deal as we can expect to have just an N channel as the number of CPU threads, this would be always small.